makefiles/boot/riotboot.mk

```diff
@@ -145,10 +145,13 @@ riotboot/slot1: $(SLOT1_RIOT_BIN)
 # Default flashing rule for bootloader + slot 0
 riotboot/flash: riotboot/flash-slot0 riotboot/flash-bootloader
 
-# make applications that use the riotboot feature default to actually using it
-# Target 'all' will generate the combined file directly.
+ifeq (,$(filter riotboot_bootloader,$(FEATURES_USED)))
```

Comment:
Why not simply ```ifneq $(riotboot, $(APPLICATION))```... ?

Reply:
I had posted this as an alternative in #12003, I discussed it with @cladmi and we though that you shouldn't look at the  `APPLICATION` variable to change build behavior. Maybe @cladmi has another argument I don't recall?

Reply:
I don't see why not? Do we expect people calling their application "riotboot"?

Anyhow, an alternative to having to add the feature everywhere would be to set "BUILDING_RIOTBOOT=1" in bootloader/riotboot/Makefile and check for that.

Reply:
Or re-use "-DRIOTBOOT" as in "ifneq (,$(filter -DRIOTBOOT,$(CFLAGS)))...".

Reply:
> having to add the feature everywhere

I cannot imagine @cladmi would accept use of FEATURES for that, anyways. :wink:

Reply:
> I cannot imagine @cladmi would accept use of FEATURES for that, anyways. wink

That can be correct, he may not like any of the solutions I had proposed.

Reply:
Anyhow, before, we introduced "FLASHFILE = $(RIOTBOOT_EXTENDED_BIN) ". This is wrong in the bootloader case, as riotboot itself behaves like a normal application. So we need to get the information that we're building the bootloader into that file.
Using a feature might seem like a straightforward idea, but that forces us to make changes to completely unrelated files (board / cpu / *). Those might even be external, and they grow.
So that's a bad idea in any case.

Adding a simple "BUILDING_RIOTBOOT=1" to the riotboot makefile can do the same, without the need to add a line everywhere else.

Is makefiles/boot/riotboot.inc.mk" actually supposed to be included when building riotboot itself? Maybe we can skip including it altogether in that case.

Reply:
> Is makefiles/boot/riotboot.inc.mk" actually supposed to be included when building riotboot itself? Maybe we can skip including it altogether in that case.

I don't think it does, the target `riotboot/flash-bootloader` does though. But we would still have to change the trigger for including `makefiles/boot/riotboot.inc.mk` since right now it is still `FEATURES_USED += riotboot`. It would come down to the same kind of solutions but done elsewhere.  But in this case we would avoid including uneeded targets so your approach sounds better to me.

Reply:
> Adding a simple "BUILDING_RIOTBOOT=1" to the riotboot makefile can do the same, without the need to add a line everywhere else.

Ok, this does sound more simple. I still don't like calling it `RIOTBOOT`, but I know you don't agree with this as stated in #11940. So I'll change it to your suggestion.